### DISCUSSION TITLE
Is sasl-dart required for Quarto operation?

### DISCUSSION THREAD

**Discussion by yurivict** [quarto-dev/quarto-cli#10556] [Q&A]
### Description

Is sass-dart only required for binary executable generation?

Can libsass https://sass-lang.com/libsass (https://github.com/sass/libsass) substitute sasl-dart for the purposes of Quarto?

Motivation: I am trying to create the FreeBSD port for Quarto, but the sasl-dart dependency depends in turn on Dart which seems to be hard to port. I am trying to understand how necessary sasl-dart is for Quarto.

**Comment by cscheid** [comment]
I think you do need the binary to make Quarto work as is (because I imagine that the CLI arguments aren't equivalent).

You could attempt a fork and fix, but we would prefer for that to be first merged to Quarto instead of going directly to FreeBSD. Otherwise, we wouldn't be able to offer any support, and SCSS compilation is a large part of our HTML formats; it would be very important to get that perfectly right.

For what's worth, this _is_ a part of `quarto-cli` that is pretty well-architected and well-factored. All invocations go through `dartCommand` (which admittedly isn't the best name for what it does), defined here:

https://github.com/quarto-dev/quarto-cli/blob/92cad4456c4e8b635979d6b39f45b507aba70edb/src/core/dart-sass.ts#L55-L103

It should be possible for you to try this out and report back.
